### PR TITLE
Revise Kedro project structure docs

### PR DESCRIPTION
**Pull request by DimedS**
## Description

Addresses #3532 

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: docs/source/get_started/kedro_concepts.md [hunk 1]
```diff
@@ -63,20 +63,53 @@ The Kedro Data Catalog is the registry of all data sources that the project can
...
+- **Log**: Allows specific logging configurations by including a `logging.yml` file inside the `conf` folder. For more information about logging customisation in Kedro, visit the [Logging Customisation Guide](https://docs.kedro.org/en/stable/logging/index.html).
+- **Docs (Sphinx)**: Adds a `docs` folder with a Sphinx documentation setup. This folder is typically used to generate technical documentation for the project.
+- **Data Folder**: Adds a `data` folder structure for managing project data. The `data` folder contains multiple subfolders to store project data. We recommend you put raw data into `raw` and move processed data to other subfolders according to the [commonly accepted data engineering convention](https://towardsdatascience.com/the-importance-of-layered-thinking-in-data-engineering-a09f685edc71).
+- **PySpark**: Adds PySpark-specific configuration files.
+- **Kedro-Viz**: Adds Kedro's native visualisation tool.
```
Comment (merelcht) [inline docs/source/get_started/kedro_concepts.md:111]: You can also mention it adds experiment tracking setup

File: docs/source/get_started/kedro_concepts.md [hunk 2]
```diff
@@ -63,20 +63,53 @@ The Kedro Data Catalog is the registry of all data sources that the project can
...
+├── conf             # Project configuration files
+├── notebooks        # Project-related Jupyter notebooks (can be used for experimental code before moving the code to src)
+└── src              # Project source code
+├── .gitignore       # Hidden file that prevents staging of unnecessary files to `git`
+├── pyproject.toml   # Identifies the project root and contains configuration information
+├── README.md        # Project README
+├── requirements.txt # Project dependencies file
 ```
 
+### Tool selection and resulting structure
```
Comment (merelcht) [inline docs/source/get_started/kedro_concepts.md:101]: You can also add a link to the chapter where we discuss all the tools in more detail: https://docs.kedro.org/en/stable/starters/new_project_tools.html

**Review by merelcht — APPROVED** [review]
I added two small suggestions, but otherwise this looks great! 👍 ⭐

File: docs/source/get_started/kedro_concepts.md [hunk 3]
```diff
@@ -63,20 +63,53 @@ The Kedro Data Catalog is the registry of all data sources that the project can
...
+
+If you select all tools during project creation, your project structure will look like this:
+
+```
+project-dir          # Parent directory of the template
+├── conf             # Project configuration files
+├── data             # Local project data (not committed to version control)
+├── docs             # Project documentation
+├── notebooks        # Project-related Jupyter notebooks (can be used for experimental code before moving the code to src)
+└── src              # Project source code
```
Comment (astrojuanlu) [inline docs/source/get_started/kedro_concepts.md:78]:
```suggestion
├── src              # Project source code
```

File: docs/source/get_started/kedro_concepts.md [hunk 4]
```diff
@@ -63,20 +63,53 @@ The Kedro Data Catalog is the registry of all data sources that the project can
...
-├── data            # Local project data (not committed to version control)
-├── docs            # Project documentation
-├── notebooks       # Project-related Jupyter notebooks (can be used for experimental code before moving the code to src)
-├── pyproject.toml  # Identifies the project root and contains configuration information
-├── README.md       # Project README
-└── src             # Project source code
+project-dir          # Parent directory of the template
+├── conf             # Project configuration files
+├── notebooks        # Project-related Jupyter notebooks (can be used for experimental code before moving the code to src)
+└── src              # Project source code
```
Comment (astrojuanlu) [inline docs/source/get_started/kedro_concepts.md:94]:
```suggestion
├── src              # Project source code
```

File: docs/source/get_started/kedro_concepts.md [hunk 5]
```diff
@@ -63,20 +63,53 @@ The Kedro Data Catalog is the registry of all data sources that the project can
...
+- **Test (Pytest)**: Adds a `tests` folder for storing unit and integration tests, helping to maintain code quality and ensuring that changes in the codebase do not introduce bugs. For more information about testing in Kedro, visit the [Automated Testing Guide](https://docs.kedro.org/en/stable/development/automated_testing.html).
+- **Log**: Allows specific logging configurations by including a `logging.yml` file inside the `conf` folder. For more information about logging customisation in Kedro, visit the [Logging Customisation Guide](https://docs.kedro.org/en/stable/logging/index.html).
+- **Docs (Sphinx)**: Adds a `docs` folder with a Sphinx documentation setup. This folder is typically used to generate technical documentation for the project.
+- **Data Folder**: Adds a `data` folder structure for managing project data. The `data` folder contains multiple subfolders to store project data. We recommend you put raw data into `raw` and move processed data to other subfolders according to the [commonly accepted data engineering convention](https://towardsdatascience.com/the-importance-of-layered-thinking-in-data-engineering-a09f685edc71).
```
Comment (astrojuanlu) [inline docs/source/get_started/kedro_concepts.md:109]: Can we rephrase this so it's more clear that this is _Kedro's_ data engineering convention, and not some universal standard?

File: docs/source/get_started/kedro_concepts.md [hunk 6]
```diff
@@ -99,10 +132,10 @@ Use the `local` subfolder for **settings that should not be shared**, such as ac
...
-
-The `data` folder contains multiple subfolders to store project data. We recommend you put raw data into `raw` and move processed data to other subfolders according to the [commonly accepted data engineering convention](https://towardsdatascience.com/the-importance-of-layered-thinking-in-data-engineering-a09f685edc71).
-
 ### `src`
 
 This subfolder contains the project's source code.
+
+### Customising your project structure
+
+While the default Kedro structure is recommended for collaboration and standardisation, it is possible to adapt the folder structure if necessary. This flexibility allows you to tailor the project to your needs while maintaining a consistent and recognisable structure.
```
Comment (astrojuanlu) [inline docs/source/get_started/kedro_concepts.md:141]: Maybe you can complement this sentence by adding a bit more information or what _must_ be there, for example https://github.com/kedro-org/kedro/pull/3852/files#diff-0b95e768cfe29967b90e84676a5943c8da00182809104e1de48f78534db61d66R17-R27

**Review by astrojuanlu — COMMENTED** [review]
Great job @DimedS ! Added a few comments

**Review by astrojuanlu — APPROVED** [review]
LGTM!

**Pull request merged**